doc/ranger.1
  .\"
  .\" Standard preamble:
  .\" ========================================================================
  .de Sp \" Vertical space (when we can't use .PP)
  .if t .sp .5v
  .if n .sp
  ..
  .de Vb \" Begin verbatim text
  .ft CW
  .nf
  .ne \\$1
  ..
  .de Ve \" End verbatim text
  .ft R
  .fi
  ..
  .\" Set up some character translations and predefined strings.  \*(-- will
  .\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
  .\" double quote, and \*(R" will give a right double quote.  \*(C+ will
  .\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
  .\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
  .\" nothing in troff, for use with C<>.
  .tr \(*W-
  .ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
  .ie n \{\
  .    ds -- \(*W-
  .    ds PI pi
  .    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
  .    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
  .    ds L" ""
  .    ds R" ""
  .    ds C` ""
  .    ds C' ""
  'br\}
  .el\{\
  .    ds -- \|\(em\|
  .    ds PI \(*p
  .    ds L" ``
  .    ds R" ''
  'br\}
  .\"
  .\" Escape single quotes in literal strings from groff's Unicode transform.
  .ie \n(.g .ds Aq \(aq
  .el       .ds Aq '
  .\"
  .\" If the F register is turned on, we'll generate index entries on stderr for
  .\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
  .\" entries marked with X<> in POD.  Of course, you'll have to process the
  .\" output yourself in some meaningful fashion.
  .ie \nF \{\
  .    de IX
  .    tm Index:\\$1\t\\n%\t"\\$2"
  ..
  .    nr % 0
  .    rr F
  .\}
  .el \{\
  .    de IX
  ..
  .\}
  .\"
  .\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
  .\" Fear.  Run.  Save yourself.  No user-serviceable parts.
  .    \" fudge factors for nroff and troff
  .if n \{\
  .    ds #H 0
  .    ds #V .8m
  .    ds #F .3m
  .    ds #[ \f1
  .    ds #] \fP
  .\}
  .if t \{\
  .    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
  .    ds #V .6m
  .    ds #F 0
  .    ds #[ \&
  .    ds #] \&
  .\}
  .    \" simple accents for nroff and troff
  .if n \{\
  .    ds ' \&
  .    ds ` \&
  .    ds ^ \&
  .    ds , \&
  .    ds ~ ~
  .    ds /
  .\}
  .if t \{\
  .    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
  .    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
  .    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
  .    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
  .    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
  .    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
  .\}
  .    \" troff and (daisy-wheel) nroff accents
  .ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
  .ds 8 \h'\*(#H'\(*b\h'-\*(#H'
  .ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
  .ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
  .ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
  .ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
  .ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
  .ds ae a\h'-(\w'a'u*4/10)'e
  .ds Ae A\h'-(\w'A'u*4/10)'E
  .    \" corrections for vroff
  .if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
  .if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
  .    \" for low resolution devices (crt and lpr)
  .if \n(.H>23 .if \n(.V>19 \
  \{\
  .    ds : e
  .    ds 8 ss
  .    ds o a
  .    ds d- d\h'-1'\(ga
  .    ds D- D\h'-1'\(hy
  .    ds th \o'bp'
  .    ds Th \o'LP'
  .    ds ae ae
  .    ds Ae AE
  .\}
  .rm #[ #] #H #V #F C
  .\" ========================================================================
  .\"
  .IX Title "RANGER 1"
  .TH RANGER 1 "ranger-1.4.3" "09/29/2011" "ranger manual"
  .\" For nroff, turn off justification.  Always turn off hyphenation; it makes
  .\" way too many mistakes in technical documents.
  .if n .ad l
  .nh
  .SH "NAME"
  ranger \- visual file manager
  .SH "SYNOPSIS"
  .IX Header "SYNOPSIS"
  \&\fBranger\fR [\fIoptions\fR] [\fIpath/filename\fR]
  .SH "DESCRIPTION"
  .IX Header "DESCRIPTION"
  Ranger is a file manager with an ncurses front-end written in Python.
  .PP
>>>>>>> master
It is designed to give you a broader overview of the file system by displaying
previews and backviews, dividing the screen into several columns.  The
key bindings are similar to those of other console programs like \fBvim\fR,
\&\fBmutt\fR or \fBncmpcpp\fR so the usage will be intuitive and efficient.
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "\fB\-\-verison\fR" 14
.IX Item "--verison"
Print the version and exit.
.IP "\fB\-h\fR, \fB\-\-help\fR" 14
.IX Item "-h, --help"
Print a list of options and exit.
.IP "\fB\-d\fR, \fB\-\-debug\fR" 14
.IX Item "-d, --debug"
Activate the debug mode: Whenever an error occurs, ranger will exit and print a
full traceback.  The default behavior is to merely print the name of the
exception in the statusbar/log and try to keep running.
.IP "\fB\-c\fR, \fB\-\-clean\fR" 14
.IX Item "-c, --clean"
Activate the clean mode:  Ranger will not access or create any configuration
files nor will it leave any traces on your system.  This is useful when your
configuration is broken, when you want to avoid clutter, etc.
.IP "\fB\-\-choosefile\fR=\fItargetfile\fR" 14
.IX Item "--choosefile=targetfile"
Allows you to pick a file with ranger.  This changes the behavior so that when
you open a file, ranger will exit and write the name of that file into
\&\fItargetfile\fR.
.IP "\fB\-\-choosedir\fR=\fItargetfile\fR" 14
.IX Item "--choosedir=targetfile"
Allows you to pick a directory with ranger.  When you exit ranger, it will
write the last visited directory into \fItargetfile\fR.
.IP "\fB\-\-copy\-config\fR=\fIfile\fR" 14
.IX Item "--copy-config=file"
Create copies of the default configuration files in your local configuration
directory.  Existing ones will not be overwritten.  Possible values: \fIall\fR,
\&\fIapps\fR, \fIcommands\fR, \fIkeys\fR, \fIoptions\fR, \fIscope\fR.
.IP "\fB\-\-list\-unused\-keys\fR" 14
.IX Item "--list-unused-keys"
List common keys which are not bound to any action in the \*(L"browser\*(R" context.
This list is not complete, you can bind any key that is supported by curses:
use the key code returned by \f(CW\*(C`getch()\*(C'\fR.
.IP "\fB\-\-fail\-unless\-c\fRd" 14
.IX Item "--fail-unless-cd"
Return the exit code 1 if ranger is used to run a file instead of used for file
browsing. (For example, \*(L"ranger \-\-fail\-unless\-cd test.txt\*(R" returns 1.)
.IP "\fB\-m\fR \fIn\fR, \fB\-\-mode\fR=\fIn\fR" 14
.IX Item "-m n, --mode=n"
When a filename is supplied, run it in mode \fIn\fR.  This has no effect unless
the execution of this file type is explicitly handled in the configuration.
.IP "\fB\-f\fR \fIflags\fR, \fB\-\-flags\fR=\fIflags\fR" 14
.IX Item "-f flags, --flags=flags"
When a filename is supplied, run it with the given \fIflags\fR to modify
behavior.  The execution of this file type is explicitly handled in the
configuration.
.SH "CONCEPTS"
.IX Header "CONCEPTS"
.SS "\s-1TAGS\s0"
.IX Subsection "TAGS"
Tags are single characters which are displayed left of a filename.  You can use
tags however you want.  Press \*(L"t\*(R" to toggle tags and \*(L"T\*(R" to remove any tags of
the selection. The default tag is an Asterisk (\*(L"*\*(R"), but you can use any tag by
typing \fI"<tagname>\fR.
.SS "\s-1SELECTION\s0"
.IX Subsection "SELECTION"
The \fIselection\fR is defined as \*(L"All marked files \s-1IF\s0 \s-1THERE\s0 \s-1ARE\s0 \s-1ANY\s0, otherwise
the current file.\*(R"  Be aware of this when using the :delete command, which
deletes all files in the selection.
.PP
You can mark files by pressing <Space>, v, etc.  A yellow \fBMrk\fR symbol at the
bottom right indicates that there are marked files in this directory.
.SS "\s-1MACROS\s0"
.IX Subsection "MACROS"
Macros can be used in commands to abbreviate things.
.PP
.Vb 5
\& %f   the highlighted file
\& %d   the path of the current directory
\& %s   the selected files in the current directory.
\& %t   all tagged files in the current directory
\& %c   the full paths of the currently copied/cut files
.Ve
.PP
The macros \f(CW%f\fR, \f(CW%d\fR and \f(CW%s\fR also have upper case variants, \f(CW%F\fR, \f(CW%D\fR and \f(CW%S\fR,
which refer to the next tab.  To refer to specific tabs, add a number in
between.  (%7s = selection of the seventh tab.)
.PP
\&\f(CW%c\fR is the only macro which ranges out of the current directory. So you may
\&\*(L"abuse\*(R" the copying function for other purposes, like diffing two files which
are in different directories:
.PP
.Vb 2
\& Yank the file A (type yy), move to the file B, then type
\& @diff %c %f
.Ve
.PP
Macros for file paths are generally shell-escaped so they can be used in the
:shell command.
.SS "\s-1PREVIEWS\s0"
.IX Subsection "PREVIEWS"
By default, only text files are previewed, but you can enable external preview
scripts by creating \fI~/.config/ranger/scope.sh\fR (see preview_script option.)
This script will then be executed each time you attempt to preview a file.
.PP
Fetch the default scope.sh (from \fIranger/data/scope.sh\fR) by running
\&\f(CW\*(C`ranger \-\-copy\-config=scope\*(C'\fR
.PP
This default script contains more documentation and calls to the programs
\&\fIlynx\fR and \fIelinks\fR for html, \fIhighlight\fR for text/code, \fIimg2txt\fR for
images, \fIatool\fR for archives, \fIpdftotext\fR for PDFs and \fImediainfo\fR for video
and audio
files.
.PP
Install these programs (just the ones you need) and scope.sh will automatically
use them.  Make sure to also have the options \*(L"use_preview_script\*(R" and
\&\*(L"preview_files\*(R" turned on.
.SS "\s-1BOOKMARKS\s0"
.IX Subsection "BOOKMARKS"
Type \fBm<key>\fR to bookmark the current directory. You can re-enter this
directory by typing \fB`<key>\fR. <key> can be any letter or digit.  Unlike vim,
both lowercase and uppercase bookmarks are persistent.
.PP
Each time you jump to a bookmark, the special bookmark at key ` will be set
to the last directory. So typing \*(L"``\*(R" gets you back to where you were before.
.PP
Bookmarks are selectable when tabbing in the :cd command.
.PP
Note: The bookmarks ' (Apostrophe) and ` (Backtick) are the same.
.SS "\s-1FLAGS\s0"
.IX Subsection "FLAGS"
Flags give you a way to modify the behaviour of the spawned process.  They are
used in the commands :open_with (key \*(L"r\*(R") and :shell (key \*(L"!\*(R").
.PP
.Vb 5
\& s   Silent mode.  Output will be discarded.
\& d   Detach the process.  (Run in background)
\& p   Redirect output to the pager
\& w   Wait for an Enter\-press when the process is done
\& c   Run the current file only, instead of the selection
.Ve
.PP
By default, all the flags are off unless specified otherwise in the \fIapps.py\fR
configuration file.  You can specify as many flags as you want.  An uppercase
flag negates the effect: \*(L"ddcccDs\*(R" is equivalent to \*(L"cs\*(R".
.PP
Examples: \f(CW\*(C`:open_with p\*(C'\fR will pipe the output of that process into
the pager.  \f(CW\*(C`:shell \-w df\*(C'\fR will run \*(L"df\*(R" and wait for you to press Enter before
switching back to ranger.
.SS "\s-1MODES\s0"
.IX Subsection "MODES"
By specifying a mode (a positive integer), you can tell ranger what to do with
a file when running it. You can specify which mode to use by typing <mode>l or
<mode><Enter> or :open_with <mode>.  The default mode is 0.
.PP
Examples: \f(CW\*(C`l\*(C'\fR (mode zero) to list the contents of an archive, \f(CW\*(C`1l\*(C'\fR (mode one)
to extract an archive.  See the \fIapps.py\fR configuration file for all programs
and modes.
.SH "KEY BINDINGS"
.IX Header "KEY BINDINGS"
Many key bindings take an additional numeric argument.  Type \fI5j\fR to move
down 5 lines, \fI2l\fR to open a file in mode 2, \fI10<Space>\fR to mark 10 files
or \fI3?\fR to read the third chapter of the documentation.
.PP
Key bindings can be changed.  Detailed instructions for this are in the
key binding configuration file at \fIranger/defaults/keys.py\fR. You can copy it
to your local configuration directory with the \-\-copy\-config option.
.SS "\s-1MAIN\s0 \s-1BINDINGS\s0"
.IX Subsection "MAIN BINDINGS"
.IP "h, j, k, l" 14
.IX Item "h, j, k, l"
Move left, down, up or right
.IP "^D or J, ^U or K" 14
.IX Item "^D or J, ^U or K"
Move a half page down, up
.IP "H, L" 14
.IX Item "H, L"
Move back and forward in the history
.IP "gg" 14
.IX Item "gg"
Move to the top
.IP "G" 14
.IX Item "G"
Move to the bottom
.IP "^R" 14
.IX Item "^R"
Reload everything
.IP "^L" 14
.IX Item "^L"
Redraw the screen
.IP "S" 14
.IX Item "S"
Open a shell in the current directory
.IP "yy" 14
.IX Item "yy"
Yank the selection to the \*(L"copy\*(R" buffer and mark them as to be copied
.IP "dd" 14
.IX Item "dd"
Cut the selection to the \*(L"copy\*(R" buffer and mark them as to be moved
.IP "pp" 14
.IX Item "pp"
Paste the files from the \*(L"copy\*(R" buffer here (by moving or copying, depending on
how they are marked.) By default, this will not overwrite existing files.  To
overwrite them, use \fIpo\fR.
.IP "m\fIX\fR" 14
.IX Item "mX"
Create a bookmark with the name \fIX\fR
.IP "`\fIX\fR" 14
.IX Item "`X"
Move to the bookmark with the name \fIX\fR
.IP "n, N" 14
.IX Item "n, N"
Find the next file.  By default, this gets you to the newest file in the
directory, but if you search something using the keys /, cm, ct, ..., it will
get you to the next found entry.
.IP "N" 14
.IX Item "N"
Find the previous file.
.IP "o\fIX\fR" 14
.IX Item "oX"
Change the sort method (like in mutt)
.IP "z\fIX\fR" 14
.IX Item "zX"
Change settings.  See the settings section for a list of settings and their
hotkey.
.IP "f" 14
.IX Item "f"
Quickly navigate by entering a part of the filename.
.IP "Space" 14
.IX Item "Space"
Mark a file.
.IP "v" 14
.IX Item "v"
Toggle the mark-status of all files, unmark all files.
.IP "V, uv" 14
.IX Item "V, uv"
Unmark all files
.IP "^V\fIdirection\fR" 14
.IX Item "^Vdirection"
Mark all files in the given direction.  Works just like d\fIdirection\fR.
.IP "u^V\fIdirection\fR" 14
.IX Item "u^Vdirection"
Unmark all files in the given direction.  Works just like d\fIdirection\fR.
.IP "/" 14
Search for files in the current directory.
.IP ":" 14
Open the console.
.SS "\s-1ADDITIONAL\s0 \s-1KEYBINDINGS\s0"
.IX Subsection "ADDITIONAL KEYBINDINGS"
.IP "g\fIN\fR" 14
.IX Item "gN"
Open a tab. N has to be a number from 0 to 9. If the tab doesn't exist yet, it
will be created.
.IP "gn, ^N" 14
.IX Item "gn, ^N"
Create a new tab.
.IP "gt, gT" 14
.IX Item "gt, gT"
Go to the next or previous tab. You can also use \s-1TAB\s0 and \s-1SHIFT+TAB\s0 instead.
.IP "gc, ^W" 14
.IX Item "gc, ^W"
Close the current tab.  The last tab cannot be closed this way.
.IP "?" 14
Opens the help screen with more key bindings and documentation
.SS "\s-1MIDNIGHT\s0 COMMANDER-LIKE \s-1BINDINGS\s0"
.IX Subsection "MIDNIGHT COMMANDER-LIKE BINDINGS"
.IP "<F1>" 14
.IX Item "<F1>"
Display Help.
.IP "<F3>" 14
.IX Item "<F3>"
Display the file.
.IP "<F4>" 14
.IX Item "<F4>"
Edit the file.
.IP "<F5>" 14
.IX Item "<F5>"
Copy the file.
.IP "<F6>" 14
.IX Item "<F6>"
Cut the file.
.IP "<F7>" 14
.IX Item "<F7>"
Open the console with \*(L":mkdir \*(R".
.IP "<F8>" 14
.IX Item "<F8>"
Prompt for deletion of the selected files.
.IP "<F10>" 14
.IX Item "<F10>"
Exit ranger.
.SS "READLINE-LIKE \s-1BINDINGS\s0 \s-1IN\s0 \s-1THE\s0 \s-1CONSOLE\s0"
.IX Subsection "READLINE-LIKE BINDINGS IN THE CONSOLE"
.IP "^B, ^F" 14
.IX Item "^B, ^F"
Move left and right (B for back, F for forward)
.IP "^P, ^N" 14
.IX Item "^P, ^N"
Move up and down (P for previous, N for Next)
.IP "^A, ^E" 14
.IX Item "^A, ^E"
Move to the start or to the end
.IP "^D" 14
.IX Item "^D"
Delete the current character.
.IP "^H" 14
.IX Item "^H"
Backspace.
.SH "MOUSE BUTTONS"
.IX Header "MOUSE BUTTONS"
.IP "Left Mouse Button" 4
.IX Item "Left Mouse Button"
Click on something and you'll move there.  To run a file, \*(L"enter\*(R" it, like a
directory, by clicking on the preview.
.IP "Right Mouse Button" 4
.IX Item "Right Mouse Button"
Enter a directory or run a file.
.IP "Scroll Wheel" 4
.IX Item "Scroll Wheel"
Scrolls up or down.  You can point at the column of the parent directory to
switch directories.
.SH "SETTINGS"
.IX Header "SETTINGS"
This section lists all built-in settings of ranger.  The valid types for the
value are in [brackets].  The hotkey to toggle the setting is in <brokets>, if
a hotkey exists.
.PP
Settings can be changed in the file \fI~/.config/ranger/options.py\fR or on the
fly with the command \fB:set option value\fR.  Examples:
 :set column_ratios (1,2,3)
 :set show_hidden=True
.IP "autosave_bookmarks [bool]" 4
.IX Item "autosave_bookmarks [bool]"
Save bookmarks (used with mX and `X) instantly?  This helps to synchronize
bookmarks between multiple ranger instances but leads to *slight* performance
loss.  When false, bookmarks are saved when ranger is exited.
.IP "collapse_preview [bool] <zc>" 4
.IX Item "collapse_preview [bool] <zc>"
When no preview is visible, should the last column be squeezed to make use of
the whitespace?
.IP "colorscheme_overlay [function, None]" 4
.IX Item "colorscheme_overlay [function, None]"
An overlay function for colorschemes.  See the default options.py for an
explanation and an example.
.IP "colorscheme [string]" 4
.IX Item "colorscheme [string]"
Which colorscheme to use?  These colorschemes are available by default:
\&\fBdefault\fR, \fBdefault88\fR, \fBtexas\fR, \fBjungle\fR, \fBsnow\fR. Snow is monochrome,
texas and default88 use 88 colors.
.IP "column_ratios [tuple, list]" 4
.IX Item "column_ratios [tuple, list]"
How many columns are there, and what are their relative widths?  For example, a
value of (1, 1, 1) would mean 3 even sized columns. (1, 1, 1, 1, 4) means 5 columns
with the preview column being as large as the other columns combined.
.IP "dirname_in_tabs [bool]" 4
.IX Item "dirname_in_tabs [bool]"
Display the directory name in tabs?
.IP "display_size_in_main_column [bool]" 4
.IX Item "display_size_in_main_column [bool]"
Display the file size in the main column?
.IP "display_size_in_status_bar [bool]" 4
.IX Item "display_size_in_status_bar [bool]"
Display the file size in the status bar?
.IP "display_tags_in_all_columns [bool]" 4
.IX Item "display_tags_in_all_columns [bool]"
Display tags in all columns?
.IP "draw_bookmark_borders [bool]" 4
.IX Item "draw_bookmark_borders [bool]"
Draw borders around the bookmark window?
.IP "draw_borders [bool]" 4
.IX Item "draw_borders [bool]"
Draw borders around columns?
.IP "flushinput [bool] <zi>" 4
.IX Item "flushinput [bool] <zi>"
Flush the input after each key hit?  One advantage is that when scrolling down
with \*(L"j\*(R", ranger stops scrolling instantly when you release the key.  One
disadvantage is that when you type commands blindly, some keys might get lost.
.IP "hidden_filter [regexp]" 4
.IX Item "hidden_filter [regexp]"
A regular expression pattern for files which should be hidden.
.IP "max_console_history_size [integer, None]" 4
.IX Item "max_console_history_size [integer, None]"
How many console commands should be kept in history?
.IP "max_history_size [integer, None]" 4
.IX Item "max_history_size [integer, None]"
How many directory changes should be kept in history?
.IP "mouse_enabled [bool] <zm>" 4
.IX Item "mouse_enabled [bool] <zm>"
Enable mouse input?
.IP "padding_right [bool]" 4
.IX Item "padding_right [bool]"
When collapse_preview is on and there is no preview, should there remain a
little padding on the right?  This allows you to click into that space to run
the file.
.IP "preview_directories [bool] <zP>" 4
.IX Item "preview_directories [bool] <zP>"
Preview directories in the preview column?
.IP "preview_files [bool] <zp>" 4
.IX Item "preview_files [bool] <zp>"
Preview files in the preview column?
.IP "preview_script [string, None]" 4
.IX Item "preview_script [string, None]"
Which script should handle generating previews?  If the file doesn't exist, or
use_preview_script is off, ranger will handle previews itself by just printing
the content.
.IP "save_console_history [bool]" 4
.IX Item "save_console_history [bool]"
Should the console history be saved on exit?  If disabled, the console history
is reset when you restart ranger.
.IP "scroll_offset [integer]" 4
.IX Item "scroll_offset [integer]"
Try to keep this much space between the top/bottom border when scrolling.
.IP "shorten_title [integer, bool]" 4
.IX Item "shorten_title [integer, bool]"
Trim the title of the window if it gets long?  The number defines how many
directories are displayed at once, False turns off this feature.
.IP "show_cursor [bool]" 4
.IX Item "show_cursor [bool]"
Always show the terminal cursor?
.IP "show_hidden_bookmarks [bool]" 4
.IX Item "show_hidden_bookmarks [bool]"
Show dotfiles in the bookmark preview window? (Type ')
.IP "show_hidden [bool] <zh>, <^H>" 4
.IX Item "show_hidden [bool] <zh>, <^H>"
Show hidden files?
.IP "sort_case_insensitive [bool] <zc>" 4
.IX Item "sort_case_insensitive [bool] <zc>"
Sort case-insensitively?  If true, \*(L"a\*(R" will be listed before \*(L"B\*(R" even though
its \s-1ASCII\s0 value is higher.
.IP "sort_directories_first [bool] <zd>" 4
.IX Item "sort_directories_first [bool] <zd>"
Sort directories first?
.IP "sort_reverse [bool] <or>" 4
.IX Item "sort_reverse [bool] <or>"
Sort reversed?
.IP "sort [string] <oa>, <ob>, <oc>, <om>, <on>, <ot>, <os>" 4
.IX Item "sort [string] <oa>, <ob>, <oc>, <om>, <on>, <ot>, <os>"
Which sorting mechanism should be used?  Choose one of \fBatime\fR, \fBbasename\fR,
\&\fBctime\fR, \fBmtime\fR, \fBnatural\fR, \fBtype\fR, \fBsize\fR
.Sp
Note: You can reverse the order by using an uppercase O in the key combination.
.IP "tilde_in_titlebar [bool]" 4
.IX Item "tilde_in_titlebar [bool]"
Abbreviate \f(CW$HOME\fR with ~ in the title bar (first line) of ranger?
.IP "unicode_ellipsis [bool]" 4
.IX Item "unicode_ellipsis [bool]"
Use a unicode \*(L"...\*(R" character instead of \*(L"~\*(R" to mark cut-off filenames?
.IP "update_title [bool]" 4
.IX Item "update_title [bool]"
Set a window title?
.IP "use_preview_script [bool] <zv>" 4
.IX Item "use_preview_script [bool] <zv>"
Use the preview script defined in the setting \fIpreview_script\fR?
.IP "xterm_alt_key [bool]" 4
.IX Item "xterm_alt_key [bool]"
Enable this if key combinations with the Alt Key don't work for you.
(Especially on xterm)
.SH "COMMANDS"
.IX Header "COMMANDS"
You can enter the commands in the console which is opened by pressing \*(L":\*(R".
Press <F1> to view help about the currently typed in command.
.IP "bulkrename" 2
.IX Item "bulkrename"
This command opens a list of selected files in an external editor.  After you
edit and save the file, it will generate a shell script which does bulk
renaming according to the changes you did in the file.
.Sp
This shell script is opened in an editor for you to review.  After you close
it, it will be executed.
.IP "cd [\fIdirectory\fR]" 2
.IX Item "cd [directory]"
The cd command changes the directory.  The command \f(CW\*(C`:cd \-\*(C'\fR is equivalent to
typing ``.
.IP "chmod \fIoctal_number\fR" 2
.IX Item "chmod octal_number"
Sets the permissions of the selection to the octal number.
.Sp
The octal number is between 000 and 777. The digits specify the permissions for
the user, the group and others.  A 1 permits execution, a 2 permits writing, a
4 permits reading.  Add those numbers to combine them. So a 7 permits
everything.
.Sp
Key bindings in the form of [\-+]<who><what> and =<octal> also exist.  For
example, \fB+ar\fR allows reading for everyone, \-ow forbids others to write and
=777 allows everything.
.Sp
See also: man 1 chmod
.IP "delete [\fIconfirmation\fR]" 2
.IX Item "delete [confirmation]"
Destroy all files in the selection with a roundhouse kick.  Ranger will ask for
a confirmation if you attempt to delete multiple (marked) files or non-empty
directories.
.Sp
When asking for confirmation, this command will only proceed if the last given
word starts with a `y'.
.IP "edit [\fIfilename\fR]" 2
.IX Item "edit [filename]"
Edit the current file or the file in the argument.
.IP "eval \fIpython_code\fR" 2
.IX Item "eval python_code"
Evaluates the python code.  `fm' is a reference to the \s-1FM\s0 instance.  To display
text, use the function `p'.
.Sp
Examples:
 :eval fm
 :eval len(fm.env.directories)
 :eval p(\*(L"Hello World!\*(R")
.IP "filter [\fIstring\fR]" 2
.IX Item "filter [string]"
Displays only the files which contain the \fIstring\fR in their basename.
.IP "find \fIpattern\fR" 2
.IX Item "find pattern"
Search files in the current directory that match the given (case-insensitive)
regular expression pattern as you type.  Once there is an unambiguous result,
it will be run immediately. (Or entered, if it's a directory.)
.IP "grep \fIpattern\fR" 2
.IX Item "grep pattern"
Looks for a string in all marked files or directories.
.IP "load_copy_buffer" 2
.IX Item "load_copy_buffer"
Load the copy buffer from \fI~/.config/ranger/copy_buffer\fR.  This can be used to
pass the list of copied files to another ranger instance.
.IP "mark \fIpattern\fR" 2
.IX Item "mark pattern"
Mark all files matching the regular expression pattern.
.IP "mkdir \fIdirname\fR" 2
.IX Item "mkdir dirname"
Creates a directory with the name \fIdirname\fR.
.IP "open_with [\fIapplication\fR] [\fIflags\fR] [\fImode\fR]" 2
.IX Item "open_with [application] [flags] [mode]"
Open the selected files with the given application, unless it is omitted, in
which case the default application is used.  \fIflags\fR are characters out of
\&\*(L"sdpcwSDPCW\*(R" and \fImode\fR is any positive integer. Their meanings are discussed
in their own sections.
.IP "quit" 2
.IX Item "quit"
Like quit!, but closes only this tab if multiple tabs are open.
.IP "quit!" 2
.IX Item "quit!"
Quit ranger.  The current directory will be bookmarked as ' so you can re-enter
it by typing `` or '' the next time you start ranger.
.IP "rename \fInewname\fR" 2
.IX Item "rename newname"
Rename the current file.  If a file with that name already exists, the renaming
will fail.  Also try the key binding A for appending something to a file name.
.IP "save_copy_buffer" 2
.IX Item "save_copy_buffer"
Save the copy buffer from \fI~/.config/ranger/copy_buffer\fR.  This can be used to
pass the list of copied files to another ranger instance.
.IP "search \fIpattern\fR" 2
.IX Item "search pattern"
Search files in the current directory that match the given (case insensitive)
regular expression pattern.
.IP "search_inc \fIpattern\fR" 2
.IX Item "search_inc pattern"
Search files in the current directory that match the given (case insensitive)
regular expression pattern.  This command gets you to matching files as you
type.
.IP "set \fIoption\fR=\fIvalue\fR" 2
.IX Item "set option=value"
Assigns a new value to an option.  Valid options are listed in the settings
section.  Use tab completion to get the current value of an option, though this
doesn't work for functions and regular expressions. Valid values are:
.Sp
.Vb 8
\& None           None
\& bool           True or False
\& integer        0 or 1 or \-1 or 2 etc.
\& list           [1, 2, 3]
\& tuple          1, 2, 3 or (1, 2, 3)
\& function       lambda <arguments>: <expression>
\& regexp         regexp(\*(Aq<pattern>\*(Aq)
\& string         Anything
.Ve
.IP "shell [\-\fIflags\fR] \fIcommand\fR" 2
.IX Item "shell [-flags] command"
Run a shell command.  \fIflags\fR are discussed in their own section.
.IP "terminal" 2
.IX Item "terminal"
Spawns the \fIx\-terminal-emulator\fR starting in the current directory.
.IP "touch \fIfilename\fR" 2
.IX Item "touch filename"
Creates an empty file with the name \fIfilename\fR, unless it already exists.
.IP "unmark \fIpattern\fR" 2
.IX Item "unmark pattern"
Unmark all files matching a regular expression pattern.
.SH "FILES"

ranger.py
#!/usr/bin/python -O
# -*- coding: utf-8 -*-
# Copyright (C) 2009, 2010  Roman Zimbelmann <romanz@lavabit.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

# Embed a script which allows you to change the directory of the parent shell
# after you exit ranger.  Run it with the command: source ranger ranger
"""":
if [ ! -z "$1" ]; then
	tempfile='/tmp/chosendir'
	ranger="$1"
	shift
	"$ranger" --choosedir="$tempfile" "${@:-$(pwd)}"
	test -f "$tempfile" &&
	if [ "$(cat -- "$tempfile")" != "$(echo -n `pwd`)" ]; then
		cd "$(cat "$tempfile")"
		rm -f -- "$tempfile"
	fi && return 0
else
	/bin/echo 'Usage: source path/to/ranger.py path/to/ranger.py'
fi
return 1
"""

import sys
from os.path import exists, normpath, abspath

# Need to find out whether or not the flag --clean was used ASAP,
# because --clean is supposed to disable bytecode compilation
argv = sys.argv[1:sys.argv.index('--')] if '--' in sys.argv else sys.argv[1:]
sys.dont_write_bytecode = '-c' in argv or '--clean' in argv

# Don't import ./ranger when running an installed binary at /usr/bin/ranger
if exists('ranger') and '/' in normpath(__file__) and abspath('.') in sys.path:
	sys.path.remove(abspath('.'))

# Set the actual docstring
__doc__ = """Ranger - file browser for the unix terminal"""

# Don't import ./ranger when running an installed binary at /usr/bin/ranger
if exists('ranger') and '/' in normpath(__file__) and abspath('.') in sys.path:
	sys.path.remove(abspath('.'))

# Start ranger
import ranger
sys.exit(ranger.main())
